# Week 4

[Slide](../Slide_ect/Presentation%20D%20ANST%202025.pdf)

[R code](..//Slide_ect/Lecture%20D2.R), [R code in python](../Slide_ect/Lecture_code_R.ipynb)

---

## Exercise 7.5

<img src="SymPyBilleder/2025-09-22-10-11-43.png" width="750">

### What are the 3 communalities?

The communalities is described as $h_j^2 = a_j^2 + \ldots + a_jm^2$ 

From the defined $X$ we gets the $A$ matrix, so;
$$
h_1^2 = \left(\frac{2}{3}\right)^2 + \left(\frac{1}{3}\right)^2
$$

$$
h_2^2 = \left(\frac{1}{3}\right)^2 + \left(\frac{1}{3}\right)^2
$$

$$
h_3^2 = \left(\frac{-1}{3}\right)^2 + \left(\frac{2}{3}\right)^2 
$$

This is calcualted in python

In [ ]:
h1 <- (2/3)^2 + (1/3)^2
h2 <- (1/3)^2 + (1/3)^2
h3 <- (-1/3)^2 + (2/3)^2

h1
h2
h3

[1] 0.5555556

[1] 0.2222222

[1] 0.5555556

In [ ]:
install.packages("gmp")   # run once
library(gmp)

h1 <- (as.bigq(2,3))^2 + (as.bigq(1,3))^2
h2 <- (as.bigq(1,3))^2 + (as.bigq(1,3))^2
h3 <- (as.bigq(-1,3))^2 + (as.bigq(2,3))^2

h1 
h2 
h3 

Installing package into ‘/Users/dittegilsfeldt/Library/R/4.4.3/library’
(as ‘lib’ is unspecified)


Attaching package: ‘gmp’


The following objects are masked from ‘package:base’:

    %*%, apply, crossprod, matrix, tcrossprod




Big Rational ('bigq') :
[1] 5/9

Big Rational ('bigq') :
[1] 2/9

Big Rational ('bigq') :
[1] 5/9

So, 
$$
h_1^2 = 0.55 = \left(\frac{5}{9}\right)
$$

$$
h_2^2 = 0.55 = \left(\frac{2}{9}\right)
$$

$$
h_3^2 = 0.55 = \left(\frac{5}{9}\right)
$$

### What is the distribution of $X$ (mean and dispersion)?

**The Model: $X = AF + G$ with**

$F \sim \mathcal N(0, I_2)$, $G \sim \mathcal N\!\left(0,\,\mathrm{diag}\!\left(\tfrac49,\tfrac79,\tfrac49\right)\right)$, and $F \perp G$.

**The Mean:**

$$
\mathbb E[X]=A\,\mathbb E[F]+\mathbb E[G]=0.
$$

**The Covariance (dispersion):**

$$
\Sigma_X=\operatorname{Cov}(X)=A\,\operatorname{Cov}(F)\,A^\top+\operatorname{Cov}(G)
        =A A^\top+\mathrm{diag}\!\left(\tfrac49,\tfrac79,\tfrac49\right).
$$

With

$$
A=\begin{bmatrix}
\tfrac23 & \tfrac13\\[2pt]
\tfrac13 & \tfrac13\\[2pt]
-\tfrac13 & \tfrac23
\end{bmatrix},
\quad
A A^\top=\begin{bmatrix}
\tfrac59 & \tfrac13 & 0\\[2pt]
\tfrac13 & \tfrac29 & \tfrac19\\[2pt]
0 & \tfrac19 & \tfrac59
\end{bmatrix}.
$$

Add the uniques:

$$
\Sigma_X
=\begin{bmatrix}
\tfrac59 & \tfrac13 & 0\\
\tfrac13 & \tfrac29 & \tfrac19\\
0 & \tfrac19 & \tfrac59
\end{bmatrix}
+
\begin{bmatrix}
\tfrac49&0&0\\
0&\tfrac79&0\\
0&0&\tfrac49
\end{bmatrix}
=
\begin{bmatrix}
1 & \tfrac13 & 0\\
\tfrac13 & 1 & \tfrac19\\
0 & \tfrac19 & 1
\end{bmatrix}.
$$

In [ ]:
import sympy as sp

A = sp.Matrix([[sp.Rational(2,3), sp.Rational(1,3)],
               [sp.Rational(1,3), sp.Rational(1,3)],
               [sp.Rational(-1,3), sp.Rational(2,3)]])

Sigma_F = sp.eye(2)
Sigma_G = sp.diag(sp.Rational(4,9), sp.Rational(7,9), sp.Rational(4,9))

Sigma_X = A*Sigma_F*A.T + Sigma_G
Sigma_X

Matrix([
[  1, 1/3,   0],
[1/3,   1, 1/9],
[  0, 1/9,   1]])

In [ ]:
# Factor loadings
A <- matrix(c(2/3, 1/3,
              1/3, 1/3,
             -1/3, 2/3), nrow = 3, byrow = TRUE)

# Covariances
Sigma_F <- diag(2)                         
Sigma_G <- diag(c(4/9, 7/9, 4/9))          

# Mean and covariance of X
mu_X <- rep(0, 3)
Sigma_X <- A %*% Sigma_F %*% t(A) + Sigma_G
Sigma_X

[,1]          [,2]      [,3]         
[1,]  1.000000e+00 0.3333333 -1.233581e-17
[2,]  3.333333e-01 1.0000000  1.111111e-01
[3,] -1.233581e-17 0.1111111  1.000000e+00

$X$ follows the distribution:

$$
X \sim \mathcal N\!\left(0,\;
\begin{bmatrix}
1 & \tfrac13 & 0\\
\tfrac13 & 1 & \tfrac19\\
0 & \tfrac19 & 1
\end{bmatrix}\right)
$$

Each $X_i$ has variance 1; the correlations are $ \rho_{12}=1/3$, $ \rho_{13}=0$, $ \rho_{23}=1/9$. That’s the whole story.

### What are the squared multiple correlations $\rho_{1|23}^2, \, \rho_{2|13}^2$ and $\rho_{3|12}^2$ between an X-coordinate and the two other X-coordinates?

We want $\rho^2_{1|23}, \rho^2_{2|13}, \rho^2_{3|12}$. These are the squared multiple correlations between one coordinate of $X$ and the two other coordinates. For a Gaussian vector $X \sim \mathcal N(0,\Sigma)$, the squared multiple correlation of $X_i$ with the others is

$$
R_i^2 = 1 - \frac{\operatorname{Var}(X_i \mid X_{-i})}{\operatorname{Var}(X_i)}.
$$

From part (2), we had

$$
\Sigma =
\begin{bmatrix}
1 & \tfrac{1}{3} & 0 \\
\tfrac{1}{3} & 1 & \tfrac{1}{9} \\
0 & \tfrac{1}{9} & 1
\end{bmatrix}.
$$

Each diagonal element is 1, so $\operatorname{Var}(X_i)=1$.

For a multivariate normal, the conditional variance is given by the precision matrix $K = \Sigma^{-1}$:

$$
\operatorname{Var}(X_i \mid X_{-i}) = \frac{1}{K_{ii}}.
$$

So

$$
R_i^2 = 1 - \frac{1}{K_{ii}}.
$$

$$
K = \Sigma^{-1} \;=\; \frac{1}{71}
\begin{bmatrix}
80 & -27 & 3 \\
-27 & 81 & -9 \\
3 & -9 & 72
\end{bmatrix}.
$$

So the diagonal elements are

$$
K_{11} = \tfrac{80}{71}, \quad K_{22} = \tfrac{81}{71}, \quad K_{33} = \tfrac{72}{71}.
$$

In [9]:
library(MASS)

Sigma <- matrix(c(1, 1/3, 0,
                  1/3, 1, 1/9,
                  0, 1/9, 1), 3, 3, byrow = TRUE)

K <- solve(Sigma)
R2 <- 1 - 1/diag(K)

fractions(K)   # precision matrix as fractions
fractions(R2)  # squared multiple correlations

[,1]        [,2]       [,3]       
[1,]  1.12676056 -0.3802817  0.04225352
[2,] -0.38028169  1.1408451 -0.12676056
[3,]  0.04225352 -0.1267606  1.01408451

[1]  9/80 10/81  1/72

In [10]:
R2

[1] 0.11250000 0.12345679 0.01388889

$$
\begin{aligned}
\rho^2_{1|23} &= 1 - \frac{1}{K_{11}} \;=\; 1 - \frac{71}{80} \;=\; \tfrac{9}{80} \;\approx 0.1125, \\[6pt]
\rho^2_{2|13} &= 1 - \frac{1}{K_{22}} \;=\; 1 - \frac{71}{81} \;=\; \tfrac{10}{81} \;\approx 0.1235, \\[6pt]
\rho^2_{3|12} &= 1 - \frac{1}{K_{33}} \;=\; 1 - \frac{71}{72} \;=\; \tfrac{1}{72} \;\approx 0.0139.
\end{aligned}
$$

* About **11%** of the variance of $X_1$ can be explained by $X_2$ and $X_3$.
* About **12%** of the variance of $X_2$ can be explained by $X_1$ and $X_3$.
* Only about **1%** of the variance of $X_3$ can be explained by $X_1$ and $X_2$.

## Exercise 7.7

<img src="SymPyBilleder/2025-09-22-10-13-06.png" width="750">